### PR TITLE
feat(js/plugins/anthropic): add structured output support

### PR DESCRIPTION
**Pull request by cabljac**
This is a draft PR to support structured output in the anthropic plugin, to resolve https://github.com/firebase/genkit/issues/3854

**Notes:**
- Introduces [overrides](https://github.com/firebase/genkit/pull/3881/files#diff-e51a40ac250c9696142466f114f754161e7e5102c0cdb5354548b757deb272f6L33) to force the VertexAI plugin to use `^0.24.3`, otherwise the following occurs (because of the installation of the latest version of the Anthropic SDK in the Anthropic plugin):
  image

##Features

[x] Structured output support via Anthropic beta API (structured-outputs-2025-11-13)
[x] output_format parameter passed to API for both streaming and non-streaming requests
[x] toAnthropicSchema() function transforms Zod schemas to Anthropic-compatible JSON schemas
[x] isStructuredOutputEnabled() method detects when structured output should be used
[x] Model support: claude-sonnet-4-5, claude-haiku-4-5, claude-opus-4-1, claude-opus-4-5
[x] Comprehensive test suite covering happy path and edge cases
[x] Error handling for invalid configurations
[x] Backward compatible (only activates when constrained: true and format: 'json')

### CODE REVIEW

File: js/plugins/anthropic/src/runner/beta.ts [hunk 1]
```diff
@@ -349,12 +412,24 @@ export class BetaRunner extends BaseRunner<BetaRunnerTypes> {
       body.thinking = thinkingConfig as BetaMessageCreateParams['thinking'];
     }
 
-    if (request.output?.format && request.output.format !== 'text') {
-      throw new Error(
-        `Only text output format is supported for Claude models currently`
-      );
+    // Apply structured output when model supports it and constrained output is requested
+    // TODO: factor out into a helper function? and make it cleaner
```
Comment (cabljac) [inline js/plugins/anthropic/src/runner/beta.ts]: @CorieW  can you factor this out

File: js/plugins/anthropic/tests/mocks/anthropic-client.ts [hunk 2]
```diff
@@ -366,6 +366,7 @@ export function mockMessageWithContent(
 }
 
 function toBetaMessage(message: Message): BetaMessage {
+  // @ts-ignore
```
Comment (cabljac) [inline js/plugins/anthropic/tests/mocks/anthropic-client.ts]: TODO: see if we can fix this

File: js/testapps/anthropic/src/beta/basic.ts [hunk 3]
```diff
@@ -34,6 +38,7 @@ ai.defineFlow('anthropic-beta-hello', async () => {
     prompt:
       'You are Claude on the beta API. Provide a concise greeting that mentions that you are using the beta API.',
     config: { temperature: 0.6 },
+    output: {},
```
Comment (pavelgj) [inline js/testapps/anthropic/src/beta/basic.ts]: where's the schema?

File: js/plugins/anthropic/src/runner/beta.ts [hunk 4]
```diff
@@ -66,6 +66,56 @@ const BETA_UNSUPPORTED_SERVER_TOOL_BLOCK_TYPES = new Set<string>([
...
+  // 'model-context-window-exceeded-2025-08-26',
+  // 'skills-2025-10-02',
+  // 'effort-param-2025-11-24',
+  // 'advanced-tool-use-2025-11-20',
+  'structured-outputs-2025-11-13',
+];
+
+/**
+ * Transforms a JSON schema to be compatible with Anthropic's structured output requirements.
+ * Anthropic requires `additionalProperties: false` on all object types.
```
Comment (pavelgj) [inline js/plugins/anthropic/src/runner/beta.ts:95]: really? interesting...

Comment (cabljac) [reply js/plugins/anthropic/src/runner/beta.ts:95]: I believe so https://platform.claude.com/docs/en/build-with-claude/structured-outputs

File: js/testapps/anthropic/src/beta/basic.ts (hunk 3 above)
Comment (dackers86) [reply js/testapps/anthropic/src/beta/basic.ts]: Added in the latest commit. Resolving.

File: js/testapps/anthropic/src/beta/structured_output.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        format: 'json',
+        constrained: true,
+      },
+    });
+
+    const collected: string[] = [];
+    for await (const chunk of stream) {
+      if (chunk.text) {
+        collected.push(chunk.text);
+        sendChunk(chunk.text);
```
Comment (pavelgj) [inline js/testapps/anthropic/src/beta/structured_output.ts]: with structured output this must be `chunk.output`

```suggestion
        collected.push(chunk.output);
        sendChunk(chunk.output);
```

File: js/testapps/anthropic/src/beta/structured_output.ts [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+      ],
+      config: { temperature: 0.6 },
+      output: {
+        schema: z.object({
+          name: z.string(),
+          age: z.number(),
+          city: z.string(),
+        }),
+        format: 'json',
+        constrained: true,
```
Comment (pavelgj) [inline js/testapps/anthropic/src/beta/structured_output.ts]: these are not necessary (default)

File: js/testapps/anthropic/src/beta/structured_output.ts (hunk 5 above)
Comment (pavelgj) [reply js/testapps/anthropic/src/beta/structured_output.ts]: well, not "must"... but getting JSON data instead of text that you need to parse yourself is a major feature.

**Review by pavelgj — APPROVED** [review]

File: js/testapps/anthropic/src/beta/structured_output.ts [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        },
+      ],
+      config: { temperature: 0.6 },
+      output: {
+        schema: z.object({
+          name: z.string(),
+          age: z.number(),
+          city: z.string(),
+        }),
+        format: 'json',
```
Comment (cabljac) [inline js/testapps/anthropic/src/beta/structured_output.ts:70]:
```suggestion
```

File: js/testapps/anthropic/src/beta/structured_output.ts (hunk 6 above)
Comment (cabljac) [inline js/testapps/anthropic/src/beta/structured_output.ts]:
```suggestion
```

Comment (cabljac) [reply js/testapps/anthropic/src/beta/structured_output.ts]: making a note to update these in a future PR
